### ISSUE TITLE
Wrong validation error whether an invalid yml file is submitted

### ISSUE THREAD

**Issue by sebbalex** [italia/publiccode-validator#9]
Submitting an invalid yaml file will trigger erroneously `422 Unprocessable Entity` making this misleading for a validation error. 

```json
{"status":422,"message":"Validation Errors","validationErrors":[{"Key":"yaml: unmarshal errors:","Reason":""},{"Key":"  line 1: cannot unmarshal !!str `\u003c!DOCTY...` into map[interface {}]interface {}","Reason":""}]}
```

This should be cleaner.

**Comment by saniales** [comment]
what do you expect to be written instead?

**Comment by saniales** [comment]
https://github.com/italia/publiccode-validator/blob/master/src/main.go#L311

affected line

**Comment by sebbalex** [comment]
> what do you expect to be written instead? I propose a BadRequest (400), from error description
> 
> > The HyperText Transfer Protocol (HTTP) 400 Bad Request response status code indicates that the server cannot or will not process the request **due to something that is perceived to be a client error** (e.g., **malformed request syntax**, invalid request message framing, or deceptive request routing).
> 
> ```json
> {
>     "status" : 422,
>     "message" : "Validation Errors",
>     "validationErrors" : [
>         {
>             "Key" : "yaml: unmarshal errors:",
>             "Reason" : ""
>         },
>         {
>             "Key" : "  line 1: cannot unmarshal !!str `\u003c!DOCTY...` into map[interface {}]interface {}",
>             "Reason" : ""
>         }
>     ]
> }
> ```
> 
> EDIT: added formatted version of the reference error

I think this it is a good choice!

**Comment by saniales** [comment]
Going for it <3

**Comment by sebbalex** [comment]
See this: https://github.com/italia/publiccode-validator/pull/12#issuecomment-648834804

**Comment by sebbalex** [comment]
Closed by #25 
@saniales as mentioned before we did a complete redesign of this application, thanks anyway for your time :pray:
